packages/next/server/api-utils.ts

```diff
@@ -63,6 +63,26 @@ export async function apiResolver(
       )
     }
 
+    let contentLength = 0
+    const writeData = apiRes.write
+    const endResponse = apiRes.end
+    apiRes.write = (...args: any[2]) => {
+      contentLength += Buffer.byteLength(args[0])
+      return writeData.apply(apiRes, args)
+    }
+    apiRes.end = (...args: any[2]) => {
+      if (args.length && typeof args[0] !== 'function') {
+        contentLength += Buffer.byteLength(args[0])
+      }
+
+      if (contentLength >= 5 * 1024 * 1024) {
+        console.warn(
+          `API response for ${req.url} exceeds 5mb. This will cause the request to fail in a future version. https://nextjs.org/docs/messages/api-routes-body-size-limit`
```

Comment:
```suggestion
          `API response for ${req.url} exceeds 5MB. This will cause the request to fail in a future version. https://nextjs.org/docs/messages/api-routes-body-size-limit`
```